src/usr/ipmiext/ipmidcmi.C

```diff
@@ -52,6 +159,22 @@ namespace SENSOR
 
         errlHndl_t l_err = NULL;
 
+        // Power Management support check
+        bool support;
+        l_err = getPowerManagementSupportStatus(support);
+        if (l_err != NULL)
+        {
+            TRACFCOMP(g_trac_ipmi,
+                      "Failed to check Power Management support");
+            return l_err;
+        }
+        else if (!support)
+        {
+            TRACFCOMP(g_trac_ipmi,
+                      "Power Management is not supported by BMC");
+            return l_err;
```

Comment:
This code is much cleaner.  I have a couple recommended changes:

Here are 4 possible scenarios for this code:
1. unable to read capabilities (lines 167-169)
--> By returning an error here, it again forces the system into safe mode (low performance).  To be more tollerant, we could ignore the error and still try to read the power limit (as it does with the currently merged code)  I recommend keeping the trace, set support to be true and not logging an error here.  This would allow the code to attempt to read the power limit.  It it successfully reads the limit, the system will function as normal.  If it could not read a limit at that point it will still create an error and the system will end up in safe mode as expected.

2. capabilities have been read successfully but power management is not supported (lines 173-175)
--> This means that the BMC does support power limits.  HTMGT/OCC can continue operate normally (there will not be any support for a user supplied power limit).   I recommend that no error gets returned and keep o_powerLimit as 0 and set o_limitActive to false.  This will allow HTMGT to keep the system in nominal mode.

3. capabilities have been read successfully and power management is supported.
--> normal case and should work as coded.  No changes required

4. capabilities have been read, but failed to read power limit / setting.
--> The code needs to return an error to the caller to inform them that the code was unable to determine if there was a user power limit set and/or enabled.  In this case, HTMGT would need to  keep the system in safe mode, because we do not know if there is supposed to be a user power limit to enforce.  No changes required

Reply:
Thanks for the comments

1. This is a problem if we use openbmc, because now it does not support power management. When we request a power limit value, OCC is broken and hangs ...

2. Hardcoded "return NULL", o_powerLimit already set to 0 end o_limitActive to false.

Reply:
Thanks

Reply:
>     1. This is a problem if we use openbmc, because now it does not support power management. When we request a power limit value, OCC is broken and hangs.
>        After such a request, openbmc returns an erroneous cc value from command and the program moves to line 239 (for this patch). This patch is necessary to avoid this error.
>        Unfortunately, I can`t test it with the AMI BMC software stack. But if it works, these changes will make the system more reliable in both cases. Otherwise, we can use an alternate path such as #164. But I'm not sure about this solution.

I do not understand your concern.

In your last commit: If the "get_dcmi_capability_info" command fails (line 169 or 175), then power management will be disabled and the system would end up in safe mode.  The code would never try to read the "get_power_limit".

My proposal would still allow the "get_power_limit" command to be attempted even if the "get_dcmi_capability_info" command failed:

1a. If the "get_dcmi_capability_info" command fails, but the "get_power_limit" works, then we do have enough information to maintain the user requested power limit (line 200-235).  (power management will be active)

1b. If the "get_dcmi_capability_info" command fails AND the "get_power_limit" command fails, then an error will be logged (lines 239-260) and the system will go to safe mode (no power management support).  There is nothing else that can be done for this case.

Reply:
2. Ok. Done.
In the case of an error, we believe that power management is supported and still try to read the power limit.